rainier-core/src/main/scala/com/stripe/rainier/core/RVG.scala

```diff
@@ -0,0 +1,21 @@
+package com.stripe.rainier.core
+
+case class RVG[T](value: RandomVariable[Generator[T]]) {
+  def map[U](fn: T => U): RVG[U] = RVG(value.map(_.map(fn)))
+
+  def zip[U](other: RVG[U]): RVG[(T, U)] =
+    RVG(value.zip(other.value).map { case (tg, ug) => tg.zip(ug) })
+
+  def flatMapGen[U](fn: T => Generator[U]): RVG[U] =
+    RVG(value.map(_.flatMap(fn)))
+}
+
+object RVG {
+  def traverse[T](seq: Seq[RVG[T]]): RVG[Seq[T]] =
+    RVG(
+      RandomVariable
+        .traverse(seq.map(_.value))
+        .map { gs =>
```

Comment:
I noticed in other parts of the codebase as well that you tend not to write these cases as `.map(Generator.traverse(_))`. Any particular reason?

Reply:
I'm gunshy of the rules about `_` inside nested scopes so I try to avoid using them except for something like `.map(_.foo)`.